File: backend/Models/Journey.cs
using System.ComponentModel.DataAnnotations;

namespace Backend.Models;

public class Journey
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public DateTime Departure { get; set; }
    public string DepartureStationId { get; set; } = null!;
    public string DepartureStationName { get; set; } = null!;
    public DateTime Return { get; set; }
    [StringLength(3)]
    public string ReturnStationId { get; set; } = null!;
    public string ReturnStationName { get; set; } = null!;
    public double CoveredDistance { get; set; }
    public int Duration { get; set; }
}

File: frontend/src/types/journey.ts
export interface CreateJourney {
  departure: string
  departureStationId: string
  departureStationName: string
  return: string
  returnStationId: string
  returnStationName: string
  coveredDistance: number
  duration: number
}

export interface Journey extends CreateJourney {
  id: string
}

export interface GetAllJourneys {
  result: Journey[]
  totalItems: number
}

export interface ImportJourneyResponse {
  successMessage: string
  deletedData: string
}

export interface FileData {
  file: File | null
}

export interface JourneyState {
  journeys: Journey[]
  totalJourneys: number
  journey: Journey | null
  isLoading: boolean
  isError: any
  importJourneytResponse: ImportJourneyResponse | null
}
